### PR TITLE
Add fade-in-and-up animations from Nirzar's video mock.

### PR DESCRIPTION
**Pull request by montehurd**
- [x] Smoothly animate vertical fade in on all 3 welcome screens per the mock [video](https://youtu.be/BX7AXdjFuTY)
- [x] Only run each fade once so back button and language adding aren't funky

(Note: will submit other animations from the mock video as a separate PR)

It's way more smooth on device :)
untitled mov

### CODE REVIEW

**Comment by montehurd** [comment]
@nirzar hey can you pull this down and see how it looks on device to you? I think I have the timings and motion close to the video...

File: Wikipedia/Code/UIView+WMFFadeInAndUp.m [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#import "UIView+WMFFadeInAndUp.h"
+
+@implementation UIView (WMFFadeInAndUp)
+
+- (void)wmf_zeroLayerOpacity {
+    self.layer.opacity = 0.0;
+}
+
+- (void)wmf_fadeInAndUpCompletion:(void (^ __nullable)(BOOL finished))completion {
```
Comment (montehurd) [inline Wikipedia/Code/UIView+WMFFadeInAndUp.m]: I may not need this completion block for the follow-on animation ticket. If not I'll remove it.

File: Wikipedia/Code/WMFWelcomeAnalyticsViewController.m [hunk 2]
```diff
@@ -42,6 +44,22 @@ - (void)viewDidLoad {
     [self wmf_setupTransparentWelcomeNavigationBarWithBackChevron];
 }
 
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+    static dispatch_once_t onceToken;
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFWelcomeAnalyticsViewController.m]: static means "For **ALL** instances for the entire life of the application."

I think what you mean is  "For **THIS** instance for its entire life."

Usually that means an ivar as opposed to a static variable.

File: Wikipedia/Code/WMFWelcomeAnalyticsViewController.m [hunk 3]
```diff
@@ -42,6 +44,22 @@ - (void)viewDidLoad {
...
+    dispatch_once(&onceToken, ^{
+        [self.containerView wmf_zeroLayerOpacity];
+    });
+}
+
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    static dispatch_once_t onceToken;
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFWelcomeAnalyticsViewController.m]: same here

File: Wikipedia/Code/WMFWelcomeIntroductionViewController.m [hunk 4]
```diff
@@ -38,6 +40,18 @@ - (IBAction)showHowThisWorksAlert:(id)sender {
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:animated];
     [[UIApplication sharedApplication] setStatusBarStyle:UIStatusBarStyleLightContent animated:YES];
+    static dispatch_once_t onceToken;
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFWelcomeIntroductionViewController.m]: since you are repeating this logic 3 times… it seems ripe for a superclass.

Comment (montehurd) [inline Wikipedia/Code/WMFWelcomeIntroductionViewController.m]: Ah good points! Will fix...

File: Wikipedia/Code/WMFWelcomeAnalyticsViewController.m (hunk 2 above)
Comment (montehurd) [inline Wikipedia/Code/WMFWelcomeAnalyticsViewController.m]: Fixed!

File: Wikipedia/Code/WMFWelcomeAnalyticsViewController.m (hunk 3 above)
Comment (montehurd) [inline Wikipedia/Code/WMFWelcomeAnalyticsViewController.m]: Fixed!

**Comment by coreyfloyd** [comment]
:+1:

**Comment by nirzar** [comment]
:+1: looks good (not the code, the animation :)

**Pull request merged**